stripe/_http_client.py

```diff
@@ -804,3 +927,76 @@ def _handle_request_error(self, e) -> NoReturn:
 
     def close(self):
         pass
+
+
+def raise_async_client_import_error() -> Never:
+    raise ImportError(
+        (
+            "Import httpx not found. To make async http requests,"
+            "You must either install httpx or define your own"
+            "async http client by subclassing stripe.HTTPClientAsync"
+            "and setting stripe.default_http_client to an instance of it."
+        )
+    )
+
+
+class HTTPXClient(HTTPClientAsync):
+    def __init__(self, **kwargs):
+        super(HTTPXClient, self).__init__(**kwargs)
+
+        assert httpx is not None
+        self.httpx = httpx
+        self._client = httpx.AsyncClient()
+
+    def sleep(self, secs):
+        import asyncio
+
+        return asyncio.sleep(secs)
+
+    async def request(
+        self, method, url, headers, post_data=None
+    ) -> Tuple[bytes, int, Mapping[str, str]]:
+        try:
+            response = await self._client.request(
+                method, url, headers=headers, data=post_data or {}
+            )
+        except self.httpx.HTTPError as e:
+            self._handle_request_error(e)
+
+        content = response.content
+        status_code = response.status_code
+        response_headers = response.headers
+        return content, status_code, response_headers
+
+    def _handle_request_error(self, e) -> NoReturn:
```

Comment:
Can this be shared?

Reply:
It shouldn't be. `e` is a library-specific error object and the job of this function is to convert library-specific exceptions and standardize their handling. So far this function is pretty generic, but it's good for the extension point to be exposed.

Reply:
I see we do implement different messages based on e for other libraries

Reply:
Yes, I looked through https://www.python-httpx.org/exceptions/ to see if there were any exceptions that were clearly "this exception should never be retried" but I didn't see any obvious ones.

The conservative choice is to retry everything IMO. Choosing not to retry certain classes of exception is kind of an optimization.

Reply:
Maybe UnsupportedProtocol but I don't think it's worth it.